### PR TITLE
enable caches, improve tests and docs

### PR DESCRIPTION
**Pull request by artspb**
The caches will work from the first attempt when [this issue](https://youtrack.jetbrains.com/issue/IDEA-210505) will be fixed.

### CODE REVIEW

File: src/test/groovy/org/jetbrains/intellij/SearchableOptionsSpecBase.groovy [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package org.jetbrains.intellij
+
+class SearchableOptionsSpecBase extends IntelliJPluginSpecBase {
+
+    @Override
+    String getIntellijVersion() {
+        return '2019.1'
```
Comment (zolotov) [inline src/test/groovy/org/jetbrains/intellij/SearchableOptionsSpecBase.groovy]: it's better not to introduce the new version of IDE in tests in order not to increase testing time. If you need 2019.1, let's replace all usages of 2017.2.5 to it

Comment (artspb) [reply src/test/groovy/org/jetbrains/intellij/SearchableOptionsSpecBase.groovy]: Tests for searchable options use both 2017.2.5 and 2019.1. The former is needed to check that steps are disabled for old versions and the later is used to ensure that they actually work. Should I drop negative tests then?

Comment (zolotov) [reply src/test/groovy/org/jetbrains/intellij/SearchableOptionsSpecBase.groovy]: there is another old version, 14.1.4. you can use it or migrate them to 2017.2.5 and 2017.2.5 to 2019.1.

Comment (artspb) [reply src/test/groovy/org/jetbrains/intellij/SearchableOptionsSpecBase.groovy]: Done. Took a while, though...

**Pull request merged**